js/main.js

```diff
@@ -1,44 +1,60 @@
-let matsList = ''; // final matList output
-let elementAmount = {}; // user input storeage
+let collection = {};
 
 function copyMatsList() { // copies matsList to clipboard
-    collectAmount();
-    calculateTotalMats();
-    toList();
+    let elementAmount = collectAmount();
+    calculateTotalMats(elementAmount);
+    let matsList = toList();
     let copyList = document.createElement("textarea"); // creates textarea element
     document.body.appendChild(copyList); // attaches textarea element to HTML body
     copyList.value = matsList;
     copyList.select();
     document.execCommand("copy"); // copies to clipboard
     document.body.removeChild(copyList); // removes textarea from HTML body
+    // document.getElementById("textbox").innerHTML = displayList;
     reset();
 }
 
 function toList() { // converts collection Object to string matsList
-    for (let key of Object.keys(collection)) { // key = each integer in collection Object
-        if (collection[key] === 0) { // if key is 0 contiune (skips to next key)
-            continue;
-        }
-        matsList += `${collection[key]}x ${key} \n`; // sets matsList formating
+    let matsList = '';
+    let displayList = '';
+    for (let key of Object.keys(collection)) { // key = each key in collection Object
+        // if (collection[key] === 0) { if value of each key is 0 contiune (skips to next key)
+        //     continue;
+        // }
+        matsList += `${collection[key]}x ${key} \n`; // sets matsList formating for copy
+        displayList += `${collection[key]}x ${key} <br>`;
     }
+    document.getElementById("textbox").innerHTML = displayList;
+    return matsList;
 }
 
-function calculateTotalMats() { // multiplies each key quantity against user input
+function calculateTotalMats(elementAmount) { // multiplies each key quantity against user input
+    // let collection = {};
     for (let key of Object.keys(crafting)) { // key = key of crafting Object
         for (let value of crafting[key]) { // value = value associated with key from crafting Object
-            collection[value.material] += value.quantity * elementAmount[key]; // multiplies each quantity against elementAmount
+            if (elementAmount[key] === 0) {
+                continue;
+            }
+            else if (elementAmount[key] > 0) {

```

Comment:
You don't need this in an `else if`, if you do the above comment, since it will encompass all other conditions

Reply:
known